File: backend/app/schemas/approval.py
from pydantic import BaseModel, Field
from typing import Optional


class SubmitForReviewRequest(BaseModel):
    change_summary: str = Field(..., min_length=1, max_length=500)


class ApprovalRejectRequest(BaseModel):
    comment: str = Field(..., min_length=1, max_length=500)


class ApprovalRecordResponse(BaseModel):
    id: int
    template_id: int
    version_id: int
    action: str
    comment: Optional[str] = None
    operator_id: int
    operator_name: Optional[str] = None
    operated_at: Optional[str] = None


class PendingApprovalResponse(BaseModel):
    template_id: int
    template_name: str
    version_id: int
    version_number: str
    submitter_name: str
    submitted_at: Optional[str] = None

File: frontend/src/types/approval.ts
export interface ApprovalRecord {
  id: number;
  template_id: number;
  version_id: number;
  action: string;
  comment: string | null;
  operator_id: number;
  operator_name?: string;
  operated_at: string | null;
}

export interface PendingApproval {
  template_id: number;
  template_name: string;
  version_id: number;
  version_number: string;
  submitter_name: string;
  submitted_at: string | null;
}
